### PR TITLE
Add Eslint and fix its issues

### PR DESCRIPTION
**Pull request by aminya**
This adds Eslint which helps us to clean up the code.
- Parallelized some of the async functions.
- using fs-extra stat
- Used the TypeScript module system and free functions instead of a class where possible -> _This was just copy-pasting and removing `this.`. The large code diff noise isn't helpful here._
- Removed excess async when the function doesn't use await
- Added TODO for fixing the use of constructor side-effects

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,6 +47,7 @@
     "@types/tar": "^4.0.4",
     "@types/url-join": "^4.0.0",
     "@types/which": "^2.0.0",
+    "eslint-config-atomic": "^1.14.1",
```
Comment (martin31821) [inline package.json:50]: Do we need additional eslint and/or typescript compat packages here or are they listed as dependency?

File: src/lib.ts [hunk 2]
```diff
@@ -198,10 +200,12 @@ export async function defaultBuildOptions(configs: BuildOptions, nativeonly: boo
...
 
 
-export class CMakeWrapper {
-  constructor(private options: BuildOptions) {}
-
-  async runAllConfigs(): Promise<void> {
-    console.log(this.options);
-  }
-}
+// export class CMakeWrapper {
```
Comment (martin31821) [inline src/lib.ts:203]: please remove it if unused.

File: src/main.ts [hunk 3]
```diff
@@ -67,8 +68,8 @@ const DEBUG_LOG = !!process.env.CMAKETSDEBUG;
     console.log('[ DONE ]');
 
     process.stdout.write('> Building directories... ');
-    const stagingDir = resolve(join(configs.stagingDirectory, config.os, config.arch, config.runtime, dist.abi + ''));
-    const targetDir = resolve(join(configs.targetDirectory, config.os, config.arch, config.runtime, dist.abi + ''));
+    const stagingDir = resolve(join(configs.stagingDirectory, config.os, config.arch, config.runtime, `${dist.abi  }`));
```
Comment (martin31821) [inline src/main.ts:71]: whitespace before }

File: src/runtimeDistribution.ts [hunk 4]
```diff
@@ -104,7 +106,7 @@ export class RuntimeDistribution {
       ret.reject(new Error('Failed to find NODE_MODULE_VERSION macro'));
       return;
     }
-    const version = parseInt(match[1]);
+    const version = parseInt(match[1], 10);
```
Comment (martin31821) [inline src/runtimeDistribution.ts:109]: good catch

File: src/util.ts [hunk 5]
```diff
@@ -67,7 +51,7 @@ export const EXEC = (command: string): Promise<string> => {
   return new Promise((resolve, reject) => {
     exec(command, (err, stdout, stderr) => {
       if (err) {
-        reject(new Error(err.message + '\n' + (stdout || stderr)));
+        reject(new Error(`${err.message  }\n${  stdout || stderr}`));
```
Comment (martin31821) [inline src/util.ts:54]: again whitespaces?

**Review by martin31821 — COMMENTED** [review]
Please check the few comments, otherwise LGTM.

File: package.json (hunk 1 above)
Comment (aminya) [reply package.json:50]: No, this is self-contained on npm. I can however add the required npmrc config for pnpm package manager (what I use personally).

**Comment by aminya** [comment]
I addressed the review comments. @martin31821

**Comment by aminya** [comment]
I addressed the review comments. @martin31821

**Pull request merged**